src/components/Clicking/Clicking.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { bindActionCreators, Dispatch } from "redux";
+import { BoosterpackResource } from "../../api/api";
+import { Boosterpack, Reference } from "../../models";
+import { User } from "../../models/user";
+import { sendClick } from "../../store/actions/sockets";
+import { State } from "../../store/types";
+import './Clicking.css';
+import pokeballImage from './Pokeball.png';
+
+interface ClickingProps {
+    user: User | null;
+}
+
+interface ClickingDispatchProps {
+    onPokeballClick(): void;
+}
+
+class Clicking extends Component<ClickingProps & ClickingDispatchProps> {
+    render() {
+        return (
+            <div className="Clicking">
+                <Pokeball onPokeballClick={this.props.onPokeballClick}/>
+                <PokeDollars amount={this.props.user && this.props.user.pokeDollars} rate={500}/>
+            </div>
+        );
+    }
+}
+
+class Pokeball extends Component<{onPokeballClick(): void}> {
+    render() {
+        return <div><img src={pokeballImage} alt="" onClick={(e) => { this.props.onPokeballClick(); } }/></div>;
+    }
+}
+
+class PokeDollars extends Component<{amount: number | null, rate: number}> {
+    $dollars?: HTMLSpanElement | null;
+    dead?: boolean;
+    t0?: number;
+
+    estDollars(): number | null {
+        if(this.t0 == null || this.props.amount == null || this.props.rate == null) {
+            return null;
+        }
+        const dt = (Date.now() - this.t0) * 0.001;
```

Comment:
Magic number 0.001. Maybe create a constant with a descriptive name what this number actually does?